devito/types/sparse.py

```diff
@@ -151,37 +151,6 @@ def _dist_scatter_mask(self, dmap=None):
         ret[self._sparse_position] = mask
         return tuple(ret)
 
-    def _dist_subfunc_scatter_mask(self, dmap=None):
-        """
-        This method is analogous to :meth:`_dist_scatter_mask`, although
-        the mask is now suitable to index into self's SubFunctions, rather
-        than into ``self.data``.
-        """
-        return self._dist_scatter_mask(dmap=dmap)[self._sparse_position]
-
-    def _dist_gather_mask(self, dmap=None):
-        """
-        A mask to index into the ``data`` received upon returning from
-        ``self._dist_alltoall``. This mask creates a new data array in which
-        duplicate sparse data values have been discarded. The resulting data
-        array can thus be used to populate ``self.data``.
-        """
-        ret = list(self._dist_scatter_mask(dmap=dmap))
-        mask = ret[self._sparse_position]
-        inds = np.unique(mask, return_index=True)[1]
-        inds.sort()
-        ret[self._sparse_position] = inds.tolist()
-
-        return tuple(ret)
-
-    def _dist_subfunc_gather_mask(self, dmap=None):
-        """
-        This method is analogous to :meth:`_dist_subfunc_scatter_mask`, although
-        the mask is now suitable to index into self's SubFunctions, rather
-        than into ``self.data``.
-        """
-        return self._dist_gather_mask(dmap=dmap)[self._sparse_position]
```

Comment:
Not for reviewer: THis is much "safer" to just reuse the scatter mask reciprocally than trying to infer its inverse. SO gather mask is removed